File: task2/rectangle.cpp
#include "rectangle.h"

// Ввод параметров прямоугольника из файла
void Rectangle::input(std::ifstream &stream) {
  stream >> color >> a_x >> a_y >> b_x >> b_y;
}

// Случайный ввод параметров прямоугольника
void Rectangle::inputRandom() {
  a_x = Random();
  a_y = Random();
  b_x = Random();
  b_y = Random();
  color = rand() % 7;
}

// Вывод параметров прямоугольника
void Rectangle::output(std::ofstream &stream) {
  stream << "It is RECTANGLE" << '\n';
  stream << "Angles' coordinates are: a.X = " << a_x << " a.Y = " << a_y
         << ", b.X = " << b_x << ", b.Y = " << b_y << ", color = " << getColor()
         << ", perimeter = " << perimeter() << '\n'
         << '\n';
}

// Вычисление периметра прямоугольника
double Rectangle::perimeter() {
  int x;
  int y;
  if (b_x > a_x) {
    x = b_x - a_x;
  } else {
    x = a_x - b_x;
  }
  if (b_y > a_y) {
    y = b_y - a_y;
  } else {
    y = a_y - b_y;
  }
  return 2.0 * (abs(x) + abs(y));
}

File: task3/rectangle.py
import random
from color import *


class Rectangle:
    def __init__(self):
        self.a_x = 0
        self.a_y = 0
        self.b_x = 0
        self.b_y = 0
        self.color = 0

    def input(self, string_array, i):
        if i >= len(string_array) - 4:
            return 0
        self.color = int(string_array[i])
        self.a_x = int(string_array[i + 1])
        self.a_y = int(string_array[i + 2])
        self.b_x = int(string_array[i + 3])
        self.b_y = int(string_array[i + 4])
        i += 5
        return i

    def input_random(self):
        self.color = random.randint(0, 6)
        self.a_x = random.randint(1, 1000)
        self.a_y = random.randint(1, 1000)
        self.b_x = random.randint(1, 1000)
        self.b_y = random.randint(1, 1000)

    def perimeter(self):
        if self.b_x > self.a_x:
            x = self.b_x - self.a_x
        else:
            x = self.a_x - self.b_x

        if self.b_y > self.a_y:
            y = self.b_y - self.a_y
        else:
            y = self.a_y - self.b_y
        return float(2.0 * (abs(x) + abs(y)))

    # переписать вывод (добавить периметр и колор)
    def output(self, stream):
        stream.write("It is RECTANGLE\n")
        stream.write(
            'Angles\' coordinates are: a.X = {0}, a.Y = {1}, b.X = {2}, b.Y = {3},'
            ' color = {4}, perimeter = {5} \n'.format(
                self.a_x, self.a_y, self.b_x, self.b_y, get_color(self.color), self.perimeter()))
